server/src/main/java/org/elasticsearch/rest/action/admin/indices/RestGetAliasesAction.java

```diff
@@ -84,6 +83,8 @@ public RestChannelConsumer prepareRequest(final RestRequest request, final NodeC
         getAliasesRequest.indicesOptions(IndicesOptions.fromRequest(request, getAliasesRequest.indicesOptions()));
         getAliasesRequest.local(request.paramAsBoolean("local", getAliasesRequest.local()));
 
+        //we may want to move this logic to TransportGetAliasesAction but it is based on the original provided aliases, which will
+        //not always be available there (they may get replaced so retrieving request.aliases is not quite the same).
```

Comment:
Do you mean  the conflict with how x-pack is using getAliases ( #29538 )

Reply:
yea that is something that gets easily overlooked.